### PR TITLE
Cutting stable 2023-11-01 AzureSearch swagger

### PR DESCRIPTION
**Pull request by sazeesha098**
# Data Plane API - Pull Request
Cutting stable 2023-11-01 AzureSearch swagger

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):
- [ ] GA release

### Change Scope

<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document:
* Previous API Spec Doc:
* Updated paths:

### Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

### Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/85/Swagger-Suppression-Process) 
to get approval.

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling

 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications

 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links

 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

### CODE REVIEW

File: specification/search/data-plane/Azure.Search/stable/2023-11-01/searchindex.json
Comment (robertklee) [inline specification/search/data-plane/Azure.Search/stable/2023-11-01/searchindex.json:1643]: @clareluna could you check if you need to include this? I thought it might be missing so I put it here.

Comment (claremonahan) [reply specification/search/data-plane/Azure.Search/stable/2023-11-01/searchindex.json:1643]: Thanks @robertklee -- we are actually not GAing the semantic fields version of semantic ranker. I will revert this line tomorrow AM

File: specification/search/data-plane/Azure.Search/stable/2023-11-01/searchindex.json
Comment (BevLoh) [inline specification/search/data-plane/Azure.Search/stable/2023-11-01/searchindex.json:579]: add LookupDocument definition then $ref it here? See https://github.com/Azure/azure-rest-api-specs/blob/main/specification/search/data-plane/Azure.Search/preview/2023-07-01-Preview/searchindex.json

File: specification/search/data-plane/Azure.Search/stable/2023-11-01/searchservice.json
Comment (BevLoh) [inline specification/search/data-plane/Azure.Search/stable/2023-11-01/searchservice.json:9371]: PoliCheck failres here, not sure if there's a way to suppress it

File: specification/search/data-plane/Azure.Search/stable/2023-11-01/searchservice.json
Comment (BevLoh) [inline specification/search/data-plane/Azure.Search/stable/2023-11-01/searchservice.json:6229]: was this intentionally changed? 

image

Comment (sazeesha098) [reply specification/search/data-plane/Azure.Search/stable/2023-11-01/searchservice.json:6229]: Yes, this was done to match the previous preview version and the examples.

File: specification/search/data-plane/Azure.Search/stable/2023-11-01/searchservice.json
Comment (sazeesha098) [reply specification/search/data-plane/Azure.Search/stable/2023-11-01/searchservice.json:9371]: @tjprescott?

Comment (tjprescott) [reply specification/search/data-plane/Azure.Search/stable/2023-11-01/searchservice.json:9367]: Per the details, for exceptions contact vsswagger@microsoft.com

Comment (sazeesha098) [reply specification/search/data-plane/Azure.Search/stable/2023-11-01/searchservice.json:9367]: Thanks @tjprescott

**Review by weidongxu-microsoft — APPROVED** [review]
Reviewed the diff (I assume it be diff from 2023-10-01-preview ?)
https://github.com/Azure/azure-rest-api-specs/pull/25798/files/05105f14daef1457142506101ac8692baabce9d9..53de17e0b81ad7667b247952e12cb387fce6197d

LGTM

**Comment by weidongxu-microsoft** [comment]
There is a few breaking change to existing 2023-07-01-preview, which I have not context of.

Generally, no change should happen to an existing api-version.

File: specification/search/data-plane/Azure.Search/stable/2023-11-01/searchservice.json
Comment (sazeesha098) [inline specification/search/data-plane/Azure.Search/stable/2023-11-01/searchservice.json:7294]: @robertklee did you intend to make vectorIndexSize not required.

Comment (robertklee) [reply specification/search/data-plane/Azure.Search/stable/2023-11-01/searchservice.json:7294]: @sazeesha098 it shows up as required on my side. The comment in the PR looks like it's pointing to CORS options; perhaps the version you reviewed was outdated? 

I intended to add it as "required" but since it's just the response deserialization object for index and service statistics, I think it doesn't matter whether it is required or not.

File: specification/search/data-plane/Azure.Search/stable/2023-11-01/searchindex.json
Comment (alzimmermsft) [inline specification/search/data-plane/Azure.Search/stable/2023-11-01/searchindex.json:1469]: Shouldn't this be removed as support for text is added in `2023-10-01-Preview` which won't be part of this GA. This applies to `text` in the `enum` array as well.

File: specification/search/data-plane/Azure.Search/stable/2023-11-01/searchindex.json
Comment (alzimmermsft) [inline specification/search/data-plane/Azure.Search/stable/2023-11-01/searchindex.json:1474]: I thought `VectorFilterMode` wasn't being added to the GA and is only available starting with the `2023-10-01-Preview`

File: specification/search/data-plane/Azure.Search/stable/2023-11-01/searchindex.json
Comment (alzimmermsft) [inline specification/search/data-plane/Azure.Search/stable/2023-11-01/searchindex.json:1679]: Same question about whether `VectorFilterMode` is in the GA

File: specification/search/data-plane/Azure.Search/stable/2023-11-01/searchservice.json
Comment (alzimmermsft) [inline specification/search/data-plane/Azure.Search/stable/2023-11-01/searchservice.json:2682]: Isn't this missing `Edm.Single` for vector fields?

File: specification/search/data-plane/Azure.Search/stable/2023-11-01/searchservice.json
Comment (alzimmermsft) [inline specification/search/data-plane/Azure.Search/stable/2023-11-01/searchservice.json:5543]: Just want to make sure this JSON property is `algorithms` as in `2023-07-01-Preview` this used `algorithmConfigurations`

File: specification/search/data-plane/Azure.Search/stable/2023-11-01/searchservice.json
Comment (alzimmermsft) [inline specification/search/data-plane/Azure.Search/stable/2023-11-01/searchservice.json:5655]: Is this part of the GA? I thought this was being added in the `2023-10-01-Preview`

File: specification/search/data-plane/Azure.Search/stable/2023-11-01/searchservice.json
Comment (robertklee) [reply specification/search/data-plane/Azure.Search/stable/2023-11-01/searchservice.json:2682]: Vector fields are `Collection(Edm.Single)`. We don't expose single by itself right now

File: specification/search/data-plane/Azure.Search/stable/2023-11-01/searchservice.json
Comment (robertklee) [reply specification/search/data-plane/Azure.Search/stable/2023-11-01/searchservice.json:5543]: Thanks for double checking. We renamed it as part of the `2023-10-01-Preview` changes to `algorithms` since we wanted to stop appending "configurations" to everything

File: specification/search/data-plane/Azure.Search/stable/2023-11-01/searchindex.json
Comment (robertklee) [reply specification/search/data-plane/Azure.Search/stable/2023-11-01/searchindex.json:1469]: Good call, yes this is a mistake. Removing shortly. Thanks for catching this!

File: specification/search/data-plane/Azure.Search/stable/2023-11-01/searchindex.json
Comment (robertklee) [reply specification/search/data-plane/Azure.Search/stable/2023-11-01/searchindex.json:1474]: Right now the plan is to GA this feature as well

File: specification/search/data-plane/Azure.Search/stable/2023-11-01/searchindex.json
Comment (robertklee) [reply specification/search/data-plane/Azure.Search/stable/2023-11-01/searchindex.json:1679]: Right now the plan is to GA this feature as well

File: specification/search/data-plane/Azure.Search/stable/2023-11-01/searchservice.json
Comment (robertklee) [reply specification/search/data-plane/Azure.Search/stable/2023-11-01/searchservice.json:5655]: Right now the plan is to GA this feature (exhaustive knn) as well

File: specification/search/data-plane/Azure.Search/stable/2023-11-01/searchindex.json
Comment (robertklee) [reply specification/search/data-plane/Azure.Search/stable/2023-11-01/searchindex.json]: This is removed now. Thanks again for catching this! :)

**Comment by robertklee** [comment]
@farzad528 fyi, i think that we're pushing to have this merged asap. Please let us know whether the feature is in scope.

@mattmsft Could we wait until EOD on merging?

**Comment by robertklee** [comment]
@jhendrixMSFT since weidong is OOF

**Comment by jhendrixMSFT** [comment]
@mikekistler @JeffreyRichter is this one approved to merge?  It's missing the `APIStewardshipBoard-SignedOff` tag.

**Comment by mattmsft** [comment]
> @mikekistler @JeffreyRichter is this one approved to merge? It's missing the `APIStewardshipBoard-SignedOff` tag.

We went over this PR in the following review
https://github.com/Azure/azure-rest-api-specs-pr/issues/12932#issuecomment-1728120684

**Comment by JeffreyRichter** [comment]
I think this is waiting for @DominikMe to review it.

File: specification/search/data-plane/Azure.Search/stable/2023-11-01/searchservice.json
Comment (DominikMe) [inline specification/search/data-plane/Azure.Search/stable/2023-11-01/searchservice.json:5558]: We should probably use the learn. subdomain now.

```suggestion
            "url": "https://learn.microsoft.com/rest/api/searchservice/Naming-rules"
```

File: specification/search/data-plane/Azure.Search/stable/2023-11-01/searchservice.json
Comment (DominikMe) [inline specification/search/data-plane/Azure.Search/stable/2023-11-01/searchservice.json:5582]:
```suggestion
            "url": "https://learn.microsoft.com/rest/api/searchservice/Naming-rules"
```

Comment (DominikMe) [reply specification/search/data-plane/Azure.Search/stable/2023-11-01/searchservice.json:5582]: and same for all other places

**Review by DominikMe — APPROVED** [review]

File: specification/search/data-plane/Azure.Search/stable/2023-11-01/searchservice.json
Comment (sazeesha098) [reply specification/search/data-plane/Azure.Search/stable/2023-11-01/searchservice.json]: @DominikMe I have updated the url's to the new path.

**Comment by jhendrixMSFT** [comment]
Is there a plan to add an exception for the PoliCheck failure?  The flagged term looks legitimate in this context.

**Comment by BevLoh** [comment]
@jhendrixMSFT , we have a work item to get an exception for the PoliCheck failure later. Does it needed to be done during this spec process?

**Comment by jhendrixMSFT** [comment]
No, just want to make sure it's being tracked so this doesn't get flagged in future PRs.

**Comment by BevLoh** [comment]
@jhendrixMSFT , can you help merge this?

**Pull request merged**
